app/Services/TransactionReader/Adapters/PhpSpreadsheetAdapter.php

```diff
@@ -6,12 +6,21 @@
 
 use App\Services\TransactionReader\TransactionReader;
 use Generator;
+use PhpOffice\PhpSpreadsheet\Cell\Cell;
+use PhpOffice\PhpSpreadsheet\Cell\StringValueBinder;
 use PhpOffice\PhpSpreadsheet\IOFactory;
 
 class PhpSpreadsheetAdapter implements TransactionReader
 {
     public function read(string $path): Generator
     {
+        $stringValueBinder = (new StringValueBinder())->setNumericConversion(false)
+            ->setBooleanConversion(false)
+            ->setNullConversion(false)
+            ->setFormulaConversion(false);
+
+        Cell::setValueBinder($stringValueBinder);
```

Comment:
ℹ️ PhpSpreadheet casts values by default (including floats) – this is how we prevent it from doing so